diff -u xine-lib/src/input/libdvdnav/decoder.c libdvdnav/src/decoder.c
--- xine-lib/src/input/libdvdnav/decoder.c	2003-04-29 22:55:47.000000000 +0100
+++ libdvdnav/src/decoder.c	2003-04-29 20:31:36.000000000 +0100
@@ -18,7 +18,7 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA
  *
- * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
+ * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
  *
  */
 
diff -u xine-lib/src/input/libdvdnav/decoder.h libdvdnav/src/decoder.h
--- xine-lib/src/input/libdvdnav/decoder.h	2003-04-07 19:10:46.000000000 +0100
+++ libdvdnav/src/decoder.h	2003-04-27 02:26:18.000000000 +0100
@@ -18,7 +18,7 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA
  *
- * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
+ * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
  *
  */
 
diff -u xine-lib/src/input/libdvdnav/dvd_types.h libdvdnav/src/dvd_types.h
--- xine-lib/src/input/libdvdnav/dvd_types.h	2003-04-29 16:58:30.000000000 +0100
+++ libdvdnav/src/dvd_types.h	2003-04-21 14:18:06.000000000 +0100
@@ -18,7 +18,7 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA
  *
- * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
+ * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
  *
  */
 
diff -u xine-lib/src/input/libdvdnav/dvdnav.c libdvdnav/src/dvdnav.c
--- xine-lib/src/input/libdvdnav/dvdnav.c	2003-04-29 23:18:14.000000000 +0100
+++ libdvdnav/src/dvdnav.c	2003-04-29 15:22:22.000000000 +0100
@@ -17,7 +17,7 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA
  *
- * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
+ * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
  *
  */
 
@@ -70,7 +70,7 @@
   struct timeval time;
   
   /* Create a new structure */
-  fprintf(MSG_OUT, "libdvdnav: Using dvdnav version %s from http://xine.sf.net\n", VERSION);
+  fprintf(MSG_OUT, "libdvdnav: Using dvdnav version %s from http://dvd.sf.net\n", VERSION);
 
   (*dest) = NULL;
   this = (dvdnav_t*)malloc(sizeof(dvdnav_t));
diff -u xine-lib/src/input/libdvdnav/dvdnav.h libdvdnav/src/dvdnav.h
--- xine-lib/src/input/libdvdnav/dvdnav.h	2003-04-29 16:58:30.000000000 +0100
+++ libdvdnav/src/dvdnav.h	2003-04-27 02:26:18.000000000 +0100
@@ -17,7 +17,7 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA
  *
- * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
+ * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
  *
  */
 
diff -u xine-lib/src/input/libdvdnav/dvdnav_events.h libdvdnav/src/dvdnav_events.h
--- xine-lib/src/input/libdvdnav/dvdnav_events.h	2003-04-07 19:10:48.000000000 +0100
+++ libdvdnav/src/dvdnav_events.h	2003-04-27 02:26:18.000000000 +0100
@@ -17,7 +17,7 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA
  *
- * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
+ * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
  *
  */
 
diff -u xine-lib/src/input/libdvdnav/dvdnav_internal.h libdvdnav/src/dvdnav_internal.h
--- xine-lib/src/input/libdvdnav/dvdnav_internal.h	2003-04-29 16:58:30.000000000 +0100
+++ libdvdnav/src/dvdnav_internal.h	2003-04-29 23:00:56.000000000 +0100
@@ -18,7 +18,7 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA
  *
- * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
+ * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
  *
  */
 
diff -u xine-lib/src/input/libdvdnav/dvdnav_internal.h.in libdvdnav/src/dvdnav_internal.h.in
--- xine-lib/src/input/libdvdnav/dvdnav_internal.h.in	2003-04-29 16:58:30.000000000 +0100
+++ libdvdnav/src/dvdnav_internal.h.in	2003-04-28 19:23:54.000000000 +0100
@@ -17,7 +17,7 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA
  *
- * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
+ * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
  *
  */
 
diff -u xine-lib/src/input/libdvdnav/highlight.c libdvdnav/src/highlight.c
--- xine-lib/src/input/libdvdnav/highlight.c	2003-04-29 16:58:30.000000000 +0100
+++ libdvdnav/src/highlight.c	2003-04-27 02:26:18.000000000 +0100
@@ -17,7 +17,7 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA
  *
- * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
+ * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
  *
  */
 
diff -u xine-lib/src/input/libdvdnav/navigation.c libdvdnav/src/navigation.c
--- xine-lib/src/input/libdvdnav/navigation.c	2003-04-07 19:10:50.000000000 +0100
+++ libdvdnav/src/navigation.c	2003-04-06 13:45:12.000000000 +0100
@@ -17,7 +17,7 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA
  *
- * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
+ * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
  *
  */
 
diff -u xine-lib/src/input/libdvdnav/read_cache.c libdvdnav/src/read_cache.c
--- xine-lib/src/input/libdvdnav/read_cache.c	2003-04-29 22:55:52.000000000 +0100
+++ libdvdnav/src/read_cache.c	2003-04-28 17:29:22.000000000 +0100
@@ -17,7 +17,7 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA
  *
- * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
+ * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
  *
  */
 
diff -u xine-lib/src/input/libdvdnav/read_cache.h libdvdnav/src/read_cache.h
--- xine-lib/src/input/libdvdnav/read_cache.h	2002-08-08 18:49:21.000000000 +0100
+++ libdvdnav/src/read_cache.h	2002-07-12 16:46:44.000000000 +0100
@@ -17,7 +17,7 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA
  *
- * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
+ * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
  *
  */
 
diff -u xine-lib/src/input/libdvdnav/remap.c libdvdnav/src/remap.c
--- xine-lib/src/input/libdvdnav/remap.c	2003-04-29 16:58:30.000000000 +0100
+++ libdvdnav/src/remap.c	2003-04-28 17:29:22.000000000 +0100
@@ -15,7 +15,7 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA
  *
- * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
+ * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
  */
 
 #include <stdlib.h>
diff -u xine-lib/src/input/libdvdnav/remap.h libdvdnav/src/remap.h
--- xine-lib/src/input/libdvdnav/remap.h	2003-02-20 16:02:00.000000000 +0000
+++ libdvdnav/src/remap.h	2003-02-20 15:32:19.000000000 +0000
@@ -15,7 +15,7 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA
  *
- * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
+ * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
  */
 
 #ifndef __REMAP__H
diff -u xine-lib/src/input/libdvdnav/searching.c libdvdnav/src/searching.c
--- xine-lib/src/input/libdvdnav/searching.c	2003-04-29 22:55:53.000000000 +0100
+++ libdvdnav/src/searching.c	2003-04-29 15:48:53.000000000 +0100
@@ -17,7 +17,7 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA
  *
- * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
+ * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
  *
  */
 
diff -u xine-lib/src/input/libdvdnav/settings.c libdvdnav/src/settings.c
--- xine-lib/src/input/libdvdnav/settings.c	2003-03-29 13:19:09.000000000 +0000
+++ libdvdnav/src/settings.c	2003-03-27 15:12:21.000000000 +0000
@@ -17,7 +17,7 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA
  *
- * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
+ * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
  *
  */
 
diff -u xine-lib/src/input/libdvdnav/vm.c libdvdnav/src/vm.c
--- xine-lib/src/input/libdvdnav/vm.c	2003-04-29 22:55:53.000000000 +0100
+++ libdvdnav/src/vm.c	2003-04-29 16:25:20.000000000 +0100
@@ -19,7 +19,7 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA
  *
- * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
+ * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
  *
  */
 
diff -u xine-lib/src/input/libdvdnav/vm.h libdvdnav/src/vm.h
--- xine-lib/src/input/libdvdnav/vm.h	2003-04-29 22:55:54.000000000 +0100
+++ libdvdnav/src/vm.h	2003-04-29 15:49:15.000000000 +0100
@@ -19,7 +19,7 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA
  *
- * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
+ * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
  *
  */
 
diff -u xine-lib/src/input/libdvdnav/vmcmd.c libdvdnav/src/vmcmd.c
--- xine-lib/src/input/libdvdnav/vmcmd.c	2003-04-07 19:10:52.000000000 +0100
+++ libdvdnav/src/vmcmd.c	2003-04-05 17:45:45.000000000 +0100
@@ -18,7 +18,7 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA
  *
- * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
+ * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
  *
  */
 
diff -u xine-lib/src/input/libdvdnav/vmcmd.h libdvdnav/src/vmcmd.h
--- xine-lib/src/input/libdvdnav/vmcmd.h	2003-04-07 19:10:53.000000000 +0100
+++ libdvdnav/src/vmcmd.h	2003-04-27 02:26:18.000000000 +0100
@@ -18,7 +18,7 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA
  *
- * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
+ * $Id: diff_against_cvs.patch,v 1.14 2003/04/29 22:22:34 jcdutton Exp $
  *
  */
 
