diff -u -p -r1.1.1.1 types.h
--- types.h	26 Jul 2004 15:27:59 -0000	1.1.1.1
+++ types.h	26 Aug 2004 16:00:07 -0000
@@ -23,8 +23,8 @@
 ** $Id: diff_to_nosefart_cvs.patch,v 1.1 2004/08/27 19:33:37 valtri Exp $
 */
 
-#ifndef _TYPES_H_
-#define _TYPES_H_
+#ifndef _NOSEFART_TYPES_H_
+#define _NOSEFART_TYPES_H_
 
 #ifdef HAVE_CONFIG_H
 #include "config.h"
@@ -88,7 +88,7 @@ typedef  uint8    boolean;
 #define  ASSERT_MSG(msg)
 #endif
 
-#endif /* _TYPES_H_ */
+#endif /* _NOSEFART_TYPES_H_ */
 
 /*
 ** $Log: diff_to_nosefart_cvs.patch,v $
 ** Revision 1.1  2004/08/27 19:33:37  valtri
 ** MINGW32 port. Engine library and most of plugins compiles now.
 **
 ** List of some changes:
 **  - replaced some _MSC_VER by more common WIN32
 **  - define INTLDIR, remove -static flag for included intl
 **  - shared more common CFLAGS with DEBUG_CFLAGS
 **  - use WIN32_CFLAGS for all building
 **  - separate some flags into THREAD_CFLAGS_CONFIG,
 **    THREAD_CFLAGS_CONFIG and ZLIB_LIB_CONFIG for public xine-config,
 **    automatically use internal libs if necessary
 **  - don't warn about missing X for mingw and cygwin
 **  - libw32dll disabled for WIN32 (making native loader would be
 **    interesting, or porting wine code to Windows? :->)
 **  - DVB and RTP disabled for WIN32, not ported yet
 **  - fix build and fix a warning in cdda
 **  - fix build for nosefart and libfaad
 **  - implement configure option --disable-freetype
 **  - sync libxine.pc and xine-config.in
 **  - add -liberty to goom under WIN32
 **  - move original build files from included phread and zlib into archives
 **    and replace them by autotools
 **
