diff -r fe43a4a573b2 contrib/libxdg-basedir/basedir.c
--- a/contrib/libxdg-basedir/basedir.c	Fri Apr 13 22:04:05 2007 +0200
+++ b/contrib/libxdg-basedir/basedir.c	Fri Apr 13 22:05:25 2007 +0200
@@ -209,6 +209,7 @@ static char** xdgSplitPath(const char* s
 #endif
 			itemlist[i][k] = string[j];
 		}
+		itemlist[i][k] = 0;
 		/* move to next string */
 		string += j;
 		if (*string == PATH_SEPARATOR_CHAR) string++; /* skip seperator */
