diff -ru goom2k4-0/src/filters.c xine-lib/src/post/goom/filters.c
--- goom2k4-0/src/filters.c	2005-02-07 11:46:41.000000000 -0200
+++ xine-lib/src/post/goom/filters.c	2005-07-18 12:15:50.000000000 -0300
@@ -704,7 +704,7 @@
     
     data->general_speed = 0.0f;
     data->reverse = 0;
-    data->theMode = AMULETTE_MODE;
+    data->theMode = rand() % 10;
     data->waveEffect = 0;
     data->hypercosEffect = 0;
     data->vPlaneEffect = 0;
diff -ru goom2k4-0/src/goom_core.c xine-lib/src/post/goom/goom_core.c
--- goom2k4-0/src/goom_core.c	2005-02-07 11:46:41.000000000 -0200
+++ xine-lib/src/post/goom/goom_core.c	2005-07-19 12:39:22.000000000 -0300
@@ -26,6 +26,8 @@
 #include "goom_fx.h"
 #include "goomsl.h"
 
+#include "xine_internal.h"
+
 /* #define VERBOSE */
 
 #define STOP_SPEED 128
@@ -736,7 +738,12 @@
         /* affichage et swappage des buffers.. */
         goomInfo->cycle++;
         
+        /* xine: no convolve_fx */
+        /*
         goomInfo->convolve_fx.apply(&goomInfo->convolve_fx,return_val,goomInfo->outputBuf,goomInfo);
+        */
+        xine_fast_memcpy(goomInfo->outputBuf, return_val, goomInfo->screen.size * sizeof(Pixel));
+
         
         return (guint32*)goomInfo->outputBuf;
 }
diff -ru goom2k4-0/src/goom_tools.c xine-lib/src/post/goom/goom_tools.c
--- goom2k4-0/src/goom_tools.c	2005-02-07 11:46:41.000000000 -0200
+++ xine-lib/src/post/goom/goom_tools.c	2005-07-18 14:30:02.000000000 -0300
@@ -3,7 +3,6 @@
 
 GoomRandom *goom_random_init(int i) {
 	GoomRandom *grandom = (GoomRandom*)malloc(sizeof(GoomRandom));
-	srand (i);
 	grandom->pos = 1;
 	goom_random_update_array(grandom, GOOM_NB_RAND);
 	return grandom;
diff -ru goom2k4-0/src/ifs.c xine-lib/src/post/goom/ifs.c
--- goom2k4-0/src/ifs.c	2005-02-07 11:46:41.000000000 -0200
+++ xine-lib/src/post/goom/ifs.c	2005-07-19 14:20:20.000000000 -0300
@@ -503,6 +503,13 @@
 
 		for (i = 0; i < 4; i++) {
 			*tmp = (*tmp) >> cycle10;
+
+      /* xine: make it darker */
+      if( *tmp && !((*tmp) >> 1) )
+        *tmp = 1;
+      else
+        *tmp = (*tmp) >> 1;
+
 			tmp++;
 		}
 	}
diff -ru goom2k4-0/src/tentacle3d.c xine-lib/src/post/goom/tentacle3d.c
--- goom2k4-0/src/tentacle3d.c	2005-02-07 11:46:41.000000000 -0200
+++ xine-lib/src/post/goom/tentacle3d.c	2005-07-19 14:04:57.000000000 -0300
@@ -10,7 +10,7 @@
 #define D 256.0f
 
 #define nbgrid 6
-#define definitionx 15
+#define definitionx 9
 #define definitionz 45
 
 typedef struct _TENTACLE_FX_DATA {
diff -ru goom2k4-0/src/xmmx.c xine-lib/src/post/goom/xmmx.c
--- goom2k4-0/src/xmmx.c	2005-02-07 11:46:41.000000000 -0200
+++ xine-lib/src/post/goom/xmmx.c	2005-07-18 15:26:23.000000000 -0300
@@ -239,7 +239,11 @@
 
 		++loop;
 	}
-	__asm__ __volatile__ ("femms\n");
+/*#ifdef HAVE_ATHLON*/
+	__asm__ __volatile__ ("emms\n");
+/*#else
+	emms();
+#endif*/
 }
 
 #define DRAWMETHOD_PLUS_XMMX(_out,_backbuf,_col) \
@@ -387,7 +391,7 @@
 		}
 	}
 end_of_line:
-	__asm__ __volatile__ ("femms\n"); 
+	__asm__ __volatile__ ("emms\n"); 
 }
 
 #endif
