--- /usr/share/gettext/po/Makefile.in.in	2006-11-27 23:24:08.000000000 +0000
+++ ./po/Makefile.in.in	2007-05-17 14:31:43.000000000 +0100
@@ -10,7 +10,10 @@
 #
 # Origin: gettext-0.16
 
+include @srcdir@/Makefile.format
+
 PACKAGE = @PACKAGE@
+LIBNAME = @LIBNAME@
 VERSION = @VERSION@
 PACKAGE_BUGREPORT = @PACKAGE_BUGREPORT@
 
@@ -65,7 +68,7 @@
 GMOFILES = @GMOFILES@
 UPDATEPOFILES = @UPDATEPOFILES@
 DUMMYPOFILES = @DUMMYPOFILES@
-DISTFILES.common = Makefile.in.in remove-potcdate.sin \
+DISTFILES.common = Makefile.in.in remove-potcdate.sin Makefile.format \
 $(DISTFILES.common.extra1) $(DISTFILES.common.extra2) $(DISTFILES.common.extra3)
 DISTFILES = $(DISTFILES.common) Makevars POTFILES.in \
 $(POFILES) $(GMOFILES) \
@@ -136,6 +139,7 @@
 	  msgid_bugs_address='$(PACKAGE_BUGREPORT)'; \
 	fi; \
 	$(XGETTEXT) --default-domain=$(DOMAIN) --directory=$(top_srcdir) \
+	  $(XGETTEXT_FORMAT_FLAGS) $(XGETTEXT_FORMAT_FLAGS_PRIVATE) \
 	  --add-comments=TRANSLATORS: $(XGETTEXT_OPTIONS) \
 	  --files-from=$(srcdir)/POTFILES.in \
 	  --copyright-holder='$(COPYRIGHT_HOLDER)' \
