Removed the wrapper for strtok_r from public header.

Index: pthread.h
===================================================================
RCS file: /home/valtri/CVS/xine-lib/win32/contrib/pthreads/pthread.h,v
retrieving revision 1.1.1.1
diff -u -p -r1.1.1.1 pthread.h
--- pthread.h	26 Jul 2004 15:27:53 -0000	1.1.1.1
+++ pthread.h	16 Sep 2004 21:07:10 -0000
@@ -978,11 +978,6 @@ int * _errno( void );
  * actually exist on WIN32.
  */
 
-#if !defined(__MINGW32__)
-#define strtok_r( _s, _sep, _lasts ) \
-	( *(_lasts) = strtok( (_s), (_sep) ) )
-#endif /* !__MINGW32__ */
-
 #define asctime_r( _tm, _buf ) \
 	( strcpy( (_buf), asctime( (_tm) ) ), \
 	  (_buf) )
